Feature/review

chanil4129 opened the pull request.

minho-bot committed:
ItemCategory PostgreSQL

ItemCategory PostgreSQL

minho-bot committed:
ItemCategory PostgreSQL

ItemCategory PostgreSQL

chanil4129 committed:
feat: Review entity

chanil4129 committed:
feat: Review entity

chanil4129 committed:
feat: Review entity

chanil4129 committed:
feat: ReviewRepository 추가

chanil4129 committed:
feat: ReviewRepository 추가

chanil4129 committed:
feat: ReviewRepository 추가

chanil4129 committed:
feat: ReviewRepository 추가

chanil4129 committed:
feat: ReviewCreateRequest 추가

chanil4129 committed:
feat: ReviewCreateRequest 추가

chanil4129 committed:
feat: ReviewCreateRequest 추가

chanil4129 committed:
feat: ReviewResponse

chanil4129 committed:
feat: ReviewResponse

chanil4129 committed:
feat: ReviewResponse

chanil4129 committed:
feat: ReviewcreateService

chanil4129 committed:
feat: ReviewcreateService

chanil4129 committed:
feat: ReviewcreateService

chanil4129 committed:
feat: ReviewDeleteRequest

chanil4129 committed:
feat: ReviewDeleteRequest

chanil4129 committed:
feat: ReviewDeleteRequest

chanil4129 committed:
feat: ReviewDeleteService

chanil4129 committed:
feat: ReviewDeleteService

chanil4129 committed:
feat: ReviewDeleteService

minho-bot committed:
Item PostgreSQL migrate

Item PostgreSQL migrate

minho-bot committed:
Item PostgreSQL migrate

Item PostgreSQL migrate

chanil4129 committed:
feat: ReviewSearchService & ReviewUpdateService

다른 부분도 수정

chanil4129 committed:
feat: ReviewSearchService & ReviewUpdateService

다른 부분도 수정

chanil4129 committed:
feat: ReviewSearchService & ReviewUpdateService

다른 부분도 수정

chanil4129 committed:
feat: CRUD 완성 & graphql resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql
resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql
resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql
resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
feat: CRUD 완성 & graphql
resolver, service, graphql 완성 및 테스트까지 완료

chanil4129 committed:
test: ReviewCreateServiceTest 작성

chanil4129 committed:
test: ReviewCreateServiceTest 작성

chanil4129 committed:
test: ReviewCreateServiceTest 작성

chanil4129 committed:
fix: import 추가

chanil4129 committed:
fix: import 추가

chanil4129 committed:
fix: import 추가

chanil4129 committed:
fix: import 추가

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/request/ReviewUpdateRequest.kt:
```diff
@@ -0,0 +1,26 @@
+	val userId: String,
+	@field: NotBlank(message = "아이템 ID는 필수 입력 항목입니다.")
+	val itemId: String,
+	val images: List<String>,
```
이거 이미지는 변경이나 추가할 수도 있어서 그냥 String 처리하면 안됨.

Hangeulkim commented on src/main/resources/graphql/review.graphqls:
```diff
@@ -0,0 +1,41 @@
+input ReviewCreateRequest{
```
이거 코드 valid grphaql에도 추가 해야댐

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/ReviewResolver.kt:
```diff
@@ -0,0 +1,55 @@
+		return reviewCreateService.createReview(request)
+	}
+	
+	@QueryMapping
```
차라리 enum 같은 걸로 query 하나로 하는게 편하지 않을까?

Hangeulkim replied:
review 검색할 때 해당 유저와 아이템으로 동시 검색할 수도 있으니 list로 받고

chanil4129 committed:
refactor: graphql valid 추가 & enum 추가 및 쿼리 수정

chanil4129 committed:
refactor: graphql valid 추가 & enum 추가 및 쿼리 수정

chanil4129 committed:
refactor: graphql valid 추가 & enum 추가 및 쿼리 수정

chanil4129 committed:
refactor: graphql valid 추가 & enum 추가 및 쿼리 수정

chanil4129 committed:
feat: 이미지 추가

chanil4129 committed:
feat: 이미지 추가

chanil4129 committed:
feat: 이미지 추가

chanil4129 committed:
feat: 이미지 추가

JooHui-void commented on src/main/kotlin/sosteam/deamhome/domain/review/exception/ReviewIllegalArgumentIdException.kt:
```diff
@@ -0,0 +1,14 @@
+	@JvmField
+	@Suppress("INAPPLICABLE_JVM_FIELD")
+	override val message: String = "ID의 개수가 조건에 맞지 않습니다."
+) : CustomGraphQLException(errorCode, ErrorType.NOT_FOUND, message) {
```
ErrorType.BAD_REQUEST 가 더 맞을 것 같네요!

JooHui-void commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/ReviewResolver.kt:
```diff
@@ -0,0 +1,54 @@
+			USER_AND_ITEM_ID -> reviewSearchService.searchReviewByUserAndItemId(request)
+			REVIEW_ID -> listOf(reviewSearchService.searchReviewByReviewId(request))
+			USER_ID -> reviewSearchService.searchReviewsByUserId(request)
+			ITEM_ID -> reviewSearchService.searchReviewsByItemId(request)
```
enum값으로 받는거면 프론트에서 enum값을 알 수 있게 graphql에서 주석으로 알려줘야 할 것 같아요

JooHui-void commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/request/ReviewCreateRequest.kt:
```diff
@@ -0,0 +1,24 @@
+
+data class ReviewCreateRequest(
+	@field: NotBlank(message = "제목은 필수 입력 항목입니다.")
+	@field: Size(max = 10, message = "제목 최대 길이는 10 입니다.")
```
10은 너무 짧지않나..? 아닐수도 있음

chanil4129 replied:
30으로 고쳤으

JooHui-void commented on src/main/kotlin/sosteam/deamhome/domain/review/service/ReviewDeleteService.kt:
```diff
@@ -0,0 +1,17 @@
+	private val reviewRepository: ReviewRepository
+) {
+	suspend fun deleteReview(request: ReviewDeleteRequest): ResponseEntity<String> {
+		reviewRepository.deleteById(request.reviewId).awaitSingleOrNull()
+		return ResponseEntity.ok("Review 제거 성공 : `${request.reviewId}` ")
```
soft delete라 아이템 없어도 제거 성공 반환할 것 같은데 상관은 없을지도?

JooHui-void replied:
아 그리고 삭제할 때 리뷰에 달린 이미지 더 안쓰면 이미지서버에서 지워야해요!

JooHui-void commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/request/ReviewCreateRequest.kt:
```diff
@@ -0,0 +1,24 @@
+	val content: String,
+	@field: DecimalMin(value = "0")
+	@field: DecimalMax(value = "5")
+	val score: Double,
```
![image](https://github.com/S-OSTeam/shop-api/assets/69834729/02896424-60f2-487c-8dae-ebc6b0ffffb5)
피그마 보니까 리뷰 다는거 자체는 Integer이고 아이템 하나의 평균이 double같은데 확인한번 해주세요!

chanil4129 replied:
너 말이 맞는듯? 우선 Int로 고쳤어. 
근데 평균 내는거는 Item 엔티티에
```
@Transient
val reviewScore: Double
```
이런식으로 만들면 될거 같은데 맞지?

JooHui-void reviewed (changes requested):
고생하셨습니다~~ :grin:

chanil4129 committed:
refactor: pr 수정

리뷰 점수 double->int
입력 예외 NOT_FOUND->BAD_REQUEST
제목 길이 10에서 30으로 수정
리뷰 삭제할 때 이미지 삭제도 추가

chanil4129 committed:
refactor: pr 수정

리뷰 점수 double->int
입력 예외 NOT_FOUND->BAD_REQUEST
제목 길이 10에서 30으로 수정
리뷰 삭제할 때 이미지 삭제도 추가

chanil4129 committed:
refactor: pr 수정

리뷰 점수 double->int
입력 예외 NOT_FOUND->BAD_REQUEST
제목 길이 10에서 30으로 수정
리뷰 삭제할 때 이미지 삭제도 추가

chanil4129 committed:
refactor: pr 수정

리뷰 점수 double->int
입력 예외 NOT_FOUND->BAD_REQUEST
제목 길이 10에서 30으로 수정
리뷰 삭제할 때 이미지 삭제도 추가

minho-bot committed:
migrate Mongo-> postgreSQL

migrate Mongo-> postgreSQL

minho-bot committed:
migrate Mongo-> postgreSQL

migrate Mongo-> postgreSQL

minho-bot committed:
migrate Mongo-> postgreSQL

migrate Mongo-> postgreSQL

minho-bot committed:
migrate Mongo-> postgreSQL

migrate Mongo-> postgreSQL

minho-bot committed:
create CategoryProvider

create CategoryProvider

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/handler/ReviewResolver.kt:
```diff
@@ -0,0 +1,54 @@
+	): List<ReviewResponse> {
+		return when (searchType) {
+			USER_AND_ITEM_ID -> reviewSearchService.searchReviewByUserAndItemId(request)
+			REVIEW_ID -> listOf(reviewSearchService.searchReviewByReviewId(request))
```
아니 이거 그 뭐지 굳이 람다 안쓰고 그 repo에 booleanexpression 쓰면 됨미다 아니면 그냥 null 넘겨주면 자동으로 true로 가요

Hangeulkim replied:
https://www.apollographql.com/blog/how-to-search-and-filter-results-with-graphql

minho-bot committed:
주석 해제, TODO 추가

TODO 추가
- AccountStatusModifyService.updateAccountStatus
- AccountAuthResolver.signOut
- ScheduledTasks.setMemberToDormant
- LogExampleController

주석 해제하고 postgreSQL 로 변경함에 따라 로직 수정
- Cart
- WishList

MongoDB 주석만 해제
- Review

minho-bot committed:
주석 해제, TODO 추가

TODO 추가
- AccountStatusModifyService.updateAccountStatus
- AccountAuthResolver.signOut
- ScheduledTasks.setMemberToDormant
- LogExampleController

주석 해제하고 postgreSQL 로 변경함에 따라 로직 수정
- Cart
- WishList

MongoDB 주석만 해제
- Review

minho-bot committed:
delete R2DBCConfiguration class

delete R2DBCConfiguration class

minho-bot committed:
delete R2DBCConfiguration class

delete R2DBCConfiguration class

chanil4129 committed:
refactor: booleanexpression 적용

chanil4129 committed:
refactor: booleanexpression 적용

chanil4129 committed:
refactor: booleanexpression 적용

chanil4129 committed:
refactor: booleanexpression 적용

minho-bot committed:
@EnableR2dbcRepositories 삭제

중복 설정으로 삭제함

minho-bot committed:
@EnableR2dbcRepositories 삭제

중복 설정으로 삭제함

minho-bot committed:
필요 없는 상속 삭제

필요 없는 상속 삭제

minho-bot committed:
필요 없는 상속 삭제

필요 없는 상속 삭제

minho-bot committed:
rename Category to CategoryEntity

rename Category to CategoryEntity

chanil4129 committed:
refactor: 좋아요 숫자만 넘기는 걸로 변경

chanil4129 committed:
refactor: 좋아요 숫자만 넘기는 걸로 변경

chanil4129 committed:
refactor: 좋아요 숫자만 넘기는 걸로 변경

chanil4129 committed:
feat: 좋아요 기능 추가

chanil4129 committed:
feat: 좋아요 기능 추가

chanil4129 committed:
feat: 좋아요 기능 추가

chanil4129 committed:
feat: 좋아요 기능 추가

chanil4129 committed:
feat: 구매 옵션

chanil4129 committed:
feat: 구매 옵션

chanil4129 committed:
feat: 구매 옵션

chanil4129 committed:
feat: 1달 사용 후기 추가

chanil4129 committed:
feat: 1달 사용 후기 추가

chanil4129 committed:
feat: 1달 사용 후기 추가

chanil4129 committed:
feat: 1달 사용 후기 추가

chanil4129 committed:
feat: 리뷰 수정 기한 지정

chanil4129 committed:
feat: 리뷰 수정 기한 지정

chanil4129 committed:
feat: 리뷰 수정 기한 지정

chanil4129 committed:
feat: 리뷰 수정 기한 지정

chanil4129 committed:
refactor: update Request 수정

userId, itemId 제거

chanil4129 committed:
refactor: update Request 수정

userId, itemId 제거

chanil4129 committed:
refactor: update Request 수정

userId, itemId 제거

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/repository/impl/ReviewRepositoryImpl.kt:
```diff
@@ -0,0 +1,51 @@
+	private val review = QReview.review
+	
+	override fun findReviews(reviewId: String?, userId: String?, itemId: String?): Flow<Review> {
+		val predicate = BooleanBuilder()
```
만드는건 이렇게 하면 됨미다 그런데 이 로직이 재사용 되거나 엄밀히 말하면 find는 아니기에 이렁건 함수로 분리합시다

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/service/ReviewSearchService.kt:
```diff
@@ -0,0 +1,37 @@
+		val responses = mutableListOf<ReviewResponse>()
+		
+		if (request.reviewId.isNotEmpty()) {
+			request.reviewId.forEach {
```
이거 in 쓰면 됨미다 querydsl에서

Hangeulkim commented on src/main/kotlin/sosteam/deamhome/domain/review/service/ReviewUpdateService.kt:
```diff
@@ -0,0 +1,56 @@
+		val now = LocalDateTime.now()
+		
+		// 리뷰 생성 후 한달이 지났는지 확인
+		if (ChronoUnit.MONTHS.between(createdAt, now) >= 1) {
```
이런 체크 로직은 search나 validation으로

Hangeulkim replied:
한달 리뷰 쪽에는 체크 로직이 없는거 같은데

chanil4129 replied:
한달 뒤 후기랑 헷갈렸어..ㅋㅋㅋ 고쳤습니다!

chanil4129 replied:
다시 확인해보니깐 기능 명세서에
"아이템 리뷰 수정 기한 지정"라고 있던데 이거 한거였으.
맞지?

Hangeulkim commented on src/main/resources/graphql/review.graphqls:
```diff
@@ -0,0 +1,57 @@
+    reviewId: String! @NotBlank
+}
+
+input ReviewSearchRequest{
```
이거 !박으면 필수 값입니다 null이 못들어와요

chanil4129 committed:
feat: 아이템 리뷰 신고

chanil4129 committed:
feat: 아이템 리뷰 신고

chanil4129 committed:
feat: 아이템 리뷰 신고

chanil4129 committed:
feat: 아이템 리뷰 신고

chanil4129 committed:
feat: 리뷰 account & item 묶기(Test 필요!!!)

account와 review 관계 설정(리뷰)
item과 review 관계 설정(평점, 리뷰 개수)

chanil4129 committed:
feat: 리뷰 account & item 묶기(Test 필요!!!)

account와 review 관계 설정(리뷰)
item과 review 관계 설정(평점, 리뷰 개수)

chanil4129 committed:
feat: 리뷰 account & item 묶기(Test 필요!!!)

account와 review 관계 설정(리뷰)
item과 review 관계 설정(평점, 리뷰 개수)

chanil4129 committed:
feat: 리뷰 account & item 묶기(Test 필요!!!)

account와 review 관계 설정(리뷰)
item과 review 관계 설정(평점, 리뷰 개수)

chanil4129 committed:
refactor : 피드백

체크 로직 validation으로 분리
graphqls "!" 제거
querydsl의 in을 써서 코드 간결화

chanil4129 committed:
refactor : 피드백

체크 로직 validation으로 분리
graphqls "!" 제거
querydsl의 in을 써서 코드 간결화

chanil4129 committed:
refactor : 피드백

체크 로직 validation으로 분리
graphqls "!" 제거
querydsl의 in을 써서 코드 간결화

chanil4129 committed:
refactor : 피드백

체크 로직 validation으로 분리
graphqls "!" 제거
querydsl의 in을 써서 코드 간결화

minho-bot committed:
주석 수정

필요 없는 주석 삭제
CartResolver package 수정

minho-bot committed:
주석 수정

필요 없는 주석 삭제
CartResolver package 수정

chanil4129 committed:
feat: 신고 기능 추가 & 기능명세서 보고 몇개 수정 & 테스트 완료

신고 기능 추가(정지 코드는 아직 작성 못함. 아마 상철이한테 물어볼 듯?)
업데이트 할 때 1달이 지나면 수정 못하게 함

chanil4129 committed:
feat: 신고 기능 추가 & 기능명세서 보고 몇개 수정 & 테스트 완료

신고 기능 추가(정지 코드는 아직 작성 못함. 아마 상철이한테 물어볼 듯?)
업데이트 할 때 1달이 지나면 수정 못하게 함

chanil4129 committed:
feat: 신고 기능 추가 & 기능명세서 보고 몇개 수정 & 테스트 완료

신고 기능 추가(정지 코드는 아직 작성 못함. 아마 상철이한테 물어볼 듯?)
업데이트 할 때 1달이 지나면 수정 못하게 함

chanil4129 committed:
feat: 신고 기능 추가 & 기능명세서 보고 몇개 수정 & 테스트 완료

신고 기능 추가(정지 코드는 아직 작성 못함. 아마 상철이한테 물어볼 듯?)
업데이트 할 때 1달이 지나면 수정 못하게 함

chanil4129 committed:
refactor: 리뷰 생성

한달 -> 일주일

chanil4129 committed:
refactor: 리뷰 생성

한달 -> 일주일

chanil4129 committed:
refactor: 리뷰 생성

한달 -> 일주일

chanil4129 committed:
feat: 아이템 좋아요 수정 및 로그 추가

chanil4129 committed:
feat: 아이템 좋아요 수정 및 로그 추가

chanil4129 committed:
feat: 아이템 좋아요 수정 및 로그 추가

minho-bot committed:
Merge branch 'feature/category-provider' into minho

minho-bot committed:
add publicId to ItemCategoryTreeResponse

add publicId to ItemCategoryTreeResponse

chanil4129 committed:
refactor: postgresql migration

chanil4129 committed:
refactor: postgresql migration

chanil4129 committed:
Merge branch 'feature/review' of https://github.com/S-OSTeam/shop-api into feature/review

minho-bot committed:
findAllCategoriesTree 메서드 공통화

findAllCategoriesTree 메서드 공통화

minho-bot committed:
Merge branch 'develop' into feature/category-provider

Hangeulkim committed:
Merge pull request #42 from S-OSTeam/feature/category-provider

create CategoryProvider

Hangeulkim committed:
feature/graphqls-restucture

이전 구조 다시 사용
extend type으로 해결 완료

chanil4129 committed:
feat: order

chanil4129 committed:
feat: order

chanil4129 committed:
Revert "feat: order"

This reverts commit 7f9dd48f9a4118f387112056ff6df93549826f63.

chanil4129 committed:
Revert "feat: order"

This reverts commit 7f9dd48f9a4118f387112056ff6df93549826f63.

Hangeulkim committed:
이미지 목록 등록, 삭제 및 table 저장 추가

이미지 목록 등록
이미지 목록 삭제
security 무력화
이미지 저장 관련 수정

Hangeulkim committed:
Merge pull request #47 from S-OSTeam/feature/html-image-parser

이미지 목록 등록, 삭제 및 table 저장 추가

Hangeulkim committed:
repository 삭제 추가

repository 삭제 추가

Hangeulkim committed:
Merge pull request #48 from S-OSTeam/feature/html-image-parser

repository 삭제 추가

JooHui-void committed:
feat: 임시저장

JooHui-void committed:
fix: create, delete r2dbc 적용

JooHui-void committed:
fix : 위시리스트 publicId로 수정

JooHui-void committed:
feat: cartItem itemId+ userId 이용해서 변경하기

JooHui-void committed:
fix: 임시 주석 취소

Hangeulkim committed:
Merge pull request #50 from S-OSTeam/feat/#44-new

[#44] 위시리스트, 장바구니 r2dbc 맞게 변경

Hangeulkim committed:
feature/test-config

postgre는 testdb로 사용
embed posgre는 자료도 미흡하고 test 환경 세팅을 확인하기 난해함
embed redis 구축완료

Hangeulkim committed:
Merge pull request #52 from S-OSTeam/feature/test-embedded-config

feature/test-config

chanil4129 committed:
refactor: rebase 맞추기

chanil4129 committed:
Merge branch 'feature/review' of https://github.com/S-OSTeam/shop-api into feature/review
